modules/dists/PrivateDist.chpl

```diff
@@ -210,6 +210,24 @@ proc PrivateArr.dsiSerialWrite(x) {
   }
 }
 
+proc PrivateArr.doiScan(op, dom) where (rank == 1) &&
+                                  chpl__scanStateResTypesMatch(op) {
+  type resType = op.generate().type;
+  var res: [dom] resType;
+
+  var localArr: [0..numLocales-1] resType;
+
+  coforall loc in Locales do on loc do
+    localArr[here.id] = if _local then data else chpl_getPrivatizedCopy(this.type, this.pid).data;
```

Comment:
This is the line I'm least sure of, but it passes testing in all the config I tried. When accesses externally, ChapelArray's `proc this` takes care of getting privatization right, but when accessing the data internally here, `dsiAccess` doesn't give you a privatized copy so I'm doing it manually here. The part that's not obvious to me is when exactly privatization is enabled. It seems to be disabled for `--local` compilations, but there's also a compiler flag and some other conditions that make me less sure this is always right.

Reply:
I think you'd technically want to mimic what ``_isPrivatized`` from ChapelArray does and check the magic ``_privatization`` value. In practice I don't know of a case when we've ever disabled privatization, so if time is a factor I'd be OK with this going in as-is.

Reply:
Think it's worth switching to using `_isPrivatized` directly. I think that's just:

```chpl
localArr[here.id] = if _isPrivatized(this) then chpl_getPrivatizedCopy(this.type, this.pid).data else data;
```

Reply:
(FYI `--baseline` disables privatization, but it looks like even hello4-datapar-dist.chpl is broken with `--comm=gasnet --baseline` on master, so maybe I'm worried about nothing here.)